src/protocol.rs

```diff
@@ -407,6 +474,134 @@ mod tests {
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut participant =
+            Participant::<AuxInfoParticipant>::from_config(&config, auxinfo_sid, ());
+
+        // Make a message with the wrong participant to field
+        let message = participant.initialize_message();
+        let bad_receiver_pid = ParticipantIdentifier::random(&mut rng);
+        assert_ne!(participant.id(), bad_receiver_pid);
+        let bad_receiver_pid_message = Message::new(
+            message.message_type,
+            message.id(),
+            message.from(),
+            bad_receiver_pid,
+            &message.unverified_bytes,
+        );
+
+        // Make sure the participant rejects the message
+        let result = participant.process_single_message(&bad_receiver_pid_message, &mut rng);
+        assert!(result.is_err());
+        assert_eq!(
+            result.unwrap_err(),
+            InternalError::CallingApplicationMistake(CallerError::WrongMessageRecipient)
+        );
+    }
+
+    #[test]
+    fn participant_rejects_messages_with_wrong_protocol_type() {
+        let mut rng = init_testing();
+        let QUORUM_SIZE = 3;
+
+        // Set up a single valid participant
+        let config = ParticipantConfig::random(QUORUM_SIZE, &mut rng);
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut participant =
+            Participant::<AuxInfoParticipant>::from_config(&config, auxinfo_sid, ());
+
+        // Make a message with the wrong protocol type
+        let message = participant.initialize_message();
+        let bad_message_type =
+            MessageType::Keygen(crate::messages::KeygenMessageType::R1CommitHash);
+        let bad_protocol_type_message = Message::new(
+            bad_message_type,
+            message.id(),
+            message.from(),
+            message.to(),
+            &message.unverified_bytes,
+        );
+
+        // Make sure the participant rejects the message
+        let result = participant.process_single_message(&bad_protocol_type_message, &mut rng);
+        assert!(result.is_err());
+        assert_eq!(
+            result.unwrap_err(),
+            InternalError::CallingApplicationMistake(CallerError::WrongProtocol)
+        );
+    }
+
+    #[test]
+    fn participant_rejects_messages_with_wrong_sender_participant() {
+        let mut rng = init_testing();
+        let QUORUM_SIZE = 3;
+
+        // Set up a single valid participant
+        let config = ParticipantConfig::random(QUORUM_SIZE, &mut rng);
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut participant =
+            Participant::<AuxInfoParticipant>::from_config(&config, auxinfo_sid, ());
+
+        //message with the wrong protocol type
+        let message = participant.initialize_message();
+        let bad_sender_pid = ParticipantIdentifier::random(&mut rng);
+        assert_ne!(participant.id(), bad_sender_pid);
```

Comment:
nit: this isn't quite the right check. We want to make sure that the `bad_sender_pid` isn't in the list of possible senders, which is `participant.all_participants()`. You can check if that list [contains](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.contains) the bad pid.